src/main/java/org/xstefank/whitelist/WhitelistProcessing.java

```diff
@@ -0,0 +1,92 @@
+package org.xstefank.whitelist;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.xstefank.api.ContinuousIntegrationAPI;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.api.TeamCityAPI;
+import org.xstefank.model.PersistentList;
+import org.xstefank.model.Utils;
+import org.xstefank.model.yaml.FormatConfig;
+
+import static org.xstefank.model.Utils.getBooleanProperty;
+import static org.xstefank.model.Utils.WHITELIST_ENABLED;
+
+public class WhitelistProcessing {
+
+    public static final boolean IS_WHITELISTING_ENABLED = getBooleanProperty(WHITELIST_ENABLED);
+
+    private static final List<ContinuousIntegrationAPI> apis = loadApis();
+
+    private final PersistentList userList;
+    private final PersistentList adminList;
+
+    private final List<Command> commands;
+
+    public WhitelistProcessing(FormatConfig config) {
+        String dirName = System.getProperty(Utils.JBOSS_CONFIG_DIR);
+        userList = new PersistentList(dirName, Utils.USERLIST_FILE_NAME);
+        adminList = new PersistentList(dirName, Utils.ADMINLIST_FILE_NAME);
+        this.commands = getCommands(config);
+    }
+
+    public static boolean isUserEligibleForCIRun(JsonNode payload, PersistentList adminList) {
+        String username = payload.get(Utils.COMMENT).get(Utils.USER).get(Utils.LOGIN).asText();
+        return adminList.hasUsername(username);
+    }
+
+    public static String getAuthorUsernameFromIssuePayload(JsonNode payload) {
```

Comment:
maybe the last two methods can be combined? they are always being used together

Reply:
There is difference in the statements in commands where these methods are used.